### PR TITLE
vimPlugins: add vim-fzf-coauthorship

### PR DESCRIPTION
**Pull request by x10an14**
Original commit message (now amended) from the below invocation:
> vimPlugins.vim-fzf-coauthorship: init at 2021-07-14

```sh
$ GITHUB_API_TOKEN="$(yq -r '."github.com".oauth_token' ~/.config/gh/hosts.yml)" ./update.py --add "maxjacobson/vim-fzf-coauthorship"
```

###### Motivation for this change
Add the [vim-fzf-coauthorship](https://github.com/maxjacobson/vim-fzf-coauthorship/) plugin to Nix! =)

This PR's work is based on: https://github.com/NixOS/nixpkgs/pull/145110

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by x10an14** [comment]
I believe the reason this might fail is due to [lingua-franca](https://github.com/lf-lang/lingua-franca.vim/) plugin has switched from `master` to `main` branch - had to fix that locally before I got the aforementioned `./update.py` invocation to work.

**Comment by SuperSandro2000** [comment]
> I believe the reason this might fail is due to [lingua-franca](https://github.com/lf-lang/lingua-franca.vim/) plugin has switched from `master` to `main` branch - had to fix that locally before I got the aforementioned `./update.py` invocation to work.

Can you add that to this PR?

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -772,6 +772,10 @@ self: super: {
     ];
   });
 
+  vim-fzf-coauthorship = super.vim-fzf-coauthorship.overrideAttrs (old: {
+    dependencies = with self; [ vim-fzf ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:776]:
```suggestion
    dependencies = with self; [ vim-fzf ];
```

Why is this so red?

Comment (x10an14) [reply pkgs/misc/vim-plugins/overrides.nix:776]: Probably non-ascii (white)space sneaked in? If that's what you're referring to. Good catch though, will fix!

**Comment by x10an14** [comment]
> > I believe the reason this might fail is due to [lingua-franca](https://github.com/lf-lang/lingua-franca.vim/) plugin has switched from `master` to `main` branch - had to fix that locally before I got the aforementioned `./update.py` invocation to work.
> 
> Can you add that to this PR?

Yup, can do - give me a day or so.

**Comment by x10an14** [comment]
I read https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md, but still didn't understand which branch to target... Mind helping me understand if it's correct that I should target `master` branch?

I'd prefer to target `nixos-unstable`, but I understand that https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md says we should backport to target branches. 

I don't mind making a 2nd backport PR, if I only can get my head wrapped around which branch I should target in the first place...

**Comment by x10an14** [comment]
Finished a day earlier than my previous estimate.

**Comment by figsoda** [comment]
targeting master is correct here, it will automatically get to nixos-unstable once this is merged, only stable releases needs backports in which you tag the pr or manually backport targeting release-21.05 branch

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 145146` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>spacevim</li>
    <li>tests.vim.test_vim_with_vim_nix</li>
    <li>vimPlugins.lingua-franca-vim</li>
    <li>vimPlugins.vim-fzf-coauthorship</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**